code/modules/events/inflation.dm

```diff
@@ -0,0 +1,81 @@
+
+/datum/round_event/economy_inflation
+	announce_when = 1
+	startWhen = 10
+	endWhen = 15
+	/// The event will modify all vendors of this specific product type.
+	var/product_type = VENDOR_PRODUCT_TYPE_FOOD
+	/// The multiplier for increasing prices of the above.
+	var/inflation_mult
+	/// Used in the announcement and in setup to indicate when it will start.
+	var/time_to_update
+
+/datum/round_event/economy_inflation/setup()
+	. = ..()
+	product_type = pick(ALL_VENDOR_PRODUCT_TYPES)
+	var/inflation_decimal = (rand(0, 9) * 0.1)
+	inflation_mult = rand(1, 3) + inflation_decimal // example result: 3.5
+	time_to_update = rand(2, 4) // minutes
+	startWhen = time_to_update * 60 // MINUTES - these already are seconds so it'd multiply by 600
+	endWhen = time_to_update * 60 + 10 // SECONDS
+
+/datum/round_event/economy_inflation/announce()
+	/* You may be thinking 'Why the fuck would W-Y directly send a broadcast to a in-operation vessel and reveal their location to any hostiles nearby?'
+	The answer is they don't, it's a signal sent across entire sectors for every UA-affiliated vessel and colony to take note of when it passes by.
+	Colony vendors aren't updated because the colony's network has collapsed. */
+	shipwide_ai_announcement("An encrypted broadband signal from Weyland-Yutani has been recieved notifying us of sudden changes in the UA's economy during cryosleep, due to [get_random_story()], and have requested us to increase the prices of [product_type] products by [get_percentage()]%. This change will come into effect in [time_to_update] minutes.")
+
+/datum/round_event/economy_inflation/start()
+	for(var/obj/structure/machinery/vending/vending_machine as anything in GLOB.total_vending_machines)
+		if(!is_mainship_level(vending_machine.z) || vending_machine.product_type != product_type)
+			continue
+		vending_machine.speak("Prices have been updated!")
+		playsound(vending_machine, 'sound/machines/twobeep.ogg', 25, TRUE)
+		var/display_records = vending_machine.product_records + vending_machine.hidden_records + vending_machine.coin_records
+		for(var/datum/data/vending_product/product in display_records)
+			product.price *= inflation_mult
+
+/**
+ *  Returns a random backstory 'incident' as the cause for the sudden price hike.
+ */
+/datum/round_event/economy_inflation/proc/get_random_story()
+	var/list/backstories = list(,
+		"A tragic series of accidental factory explosions",
+		"public leaks of corporate secrets",
+		"a high-profile assassination",
+		"a CLF take-over of an important industrial colony",
+		"discovery of high levels of an unhealthy chemical",
+		"a public scandal involving a corporate chairman and a farm animal",
+		"large-scale colony riots",
+		"suppressed factory riots over riot responses",
+		"CLF bombing of vital industrial colonies",
+		"a tense stand-off on several mining stations with UPP insurgents",
+		"a publicized Xenomorph infestation",
+		"a succession of hostile takeovers by the Company",
```

Comment:
what company?? and why would this cause price increases

Reply:
W-Y? everyone calls it the Company its almost like youre not an aliens fan

Reply:
hostile takeovers would likely cause price increases if people worry about what that could mean